### PR TITLE
Starting work improve pathfinding

### PR DESCRIPTION
**Pull request by dteviot**
@scrawl

I suspect lines 224-228 are incorrect.
When pathgrid returns a single node, it's being discarded and actor walks directly to destination.
Needs to build a path using the pathgrid node and the destination.

I also suspect that lines 232-245 may not be needed, 
aStarSearch() should not return an empty set as we're supplying two different points for a path.
Only way I can see aStarSearch() could return an empty path is if the path data we've been given is wrong. I have yet to check to see if aStarSearch() handles that case.

### CODE REVIEW

**Comment by scrawl** [comment]
I'm having trouble following your line numbers. Are those for the previous code or for your modified code?

**Comment by dteviot** [comment]
Those are the line numbers AFTER the patch is applied.  (Note, they're existing code that I haven't removed/fixed yet.)
In the pre-PR code they are:
https://github.com/OpenMW/openmw/blob/3046795a9c1197ed7fc9766fe78de84e37da1672/apps/openmw/mwmechanics/pathfinding.cpp#L228-232
and 
https://github.com/OpenMW/openmw/blob/3046795a9c1197ed7fc9766fe78de84e37da1672/apps/openmw/mwmechanics/pathfinding.cpp#L236-249
respectively.

Incidentally, how do you get the actual code lines to show in this comment?

File: apps/openmw/mwmechanics/pathfinding.cpp [hunk 1]
```diff
@@ -51,8 +51,7 @@ namespace
                                                   const MWWorld::CellStore *cell,
                                                   const osg::Vec3f pos, int start)
     {
-        if(!grid || grid->mPoints.empty())
-            return std::pair<int, bool> (-1, false);
+        assert(grid && !grid->mPoints.empty());
```
Comment (scrawl) [inline apps/openmw/mwmechanics/pathfinding.cpp]: This change breaks consistency with getClosestPoint. Either make this change in both functions or change none of them.

**Comment by scrawl** [comment]
> When pathgrid returns a single node, it's being discarded and actor walks directly to destination.
Needs to build a path using the pathgrid node and the destination.

> I also suspect that lines 232-245 may not be needed, 
> aStarSearch() should not return an empty set as we're supplying two different points for a path.

Makes sense.

**Pull request merged**